src/pmp.c

```diff
@@ -2,19 +2,14 @@
 /* SPDX-License-Identifier: Apache-2.0 */
 
 #include <metal/cpu.h>
-#include <metal/machine.h>
+#include <metal/generated/pmp.h>
```

Comment:
I think metal/pmp.h should include generated/pmp.h and this file would include that?